deprecated/map.go

```diff
@@ -0,0 +1,22 @@
+package deprecated
+
+import "reflect"
+
+// MapGroup is an implementation of the deprecated `repeated group map`
+// syntax that older parquet writers emit.
+type MapGroup[K comparable, V any] map[K]V
+
+func (m MapGroup[K, V]) IsMapGroup() {}
+
+// This interface allows us to use reflection to determine if our
+// map is a MapGroup. Unfortuntely there's no easy way to use
+// reflect.TypeOf on a generic map.
+type isMapGroup interface {
+	IsMapGroup()
```

Comment:
Let's make this a non-exported method so we don't have to leak this detail outside of the package?